### PR TITLE
chore: upgrade react query to v5

### PR DESCRIPTION
**Pull request by tyleroooo**
Seems like nothing is broken but I didn't test everything yet.

### CODE REVIEW

**Comment by tyleroooo** [comment]
investigating: undefined is no longer supported as a valid query result value and loading status was changed to pending apparently.

File: package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
     "@reduxjs/toolkit": "^1.9.5",
     "@scure/bip32": "^1.3.0",
     "@scure/bip39": "^1.2.0",
+    "@tanstack/react-query": "^5.37.1",
```
Comment (yogurtandjam) [inline package.json:85]: 🔥 🔥  LETS GOOOOOOO

**Comment by yogurtandjam** [comment]
this might take some time to fully test. i'll go through as many flows as possible when i'm in the office

File: src/hooks/useAccountBalance.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export const useAccountBalance = ({
 
   const cosmosQuery = useQuery({
     enabled: Boolean(isCosmosChain && dydxAddress && bech32AddrPrefix && rpc && addressOrDenom),
-    queryKey: `accountBalances_${chainId}_${addressOrDenom}`,
+    queryKey: [`accountBalances`, chainId, addressOrDenom],
```
Comment (aforaleka) [inline src/hooks/useAccountBalance.ts:83]: maybe nit?
```suggestion
    queryKey: ['accountBalances', chainId, addressOrDenom],
```

File: src/lib/asyncUtils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { log } from '@/lib/telemetry';
+
+export function wrapAndLogError<T>(
+  fn: () => Promise<T>,
+  logId: string,
+  // Usually we rethrow the error. If this is true, we swallow the error and return undefined.
+  errorReturnUndefined?: boolean
```
Comment (aforaleka) [inline src/lib/asyncUtils.ts:7]: should we default this to true? it seems like most of the existing use cases are swallowing the errors by default

File: src/hooks/usePerpetualMarketsStats.ts [hunk 4]
```diff
@@ -23,19 +23,16 @@ export const usePerpetualMarketsStats = () => {
 
   const { data } = useQuery({
     queryKey: ['chain-revenue', startDate.toISOString(), endDate.toISOString()],
-    queryFn: () => {
-      try {
-        return getChainRevenue({
+    queryFn: wrapAndLogError(
```
Comment (aforaleka) [inline src/hooks/usePerpetualMarketsStats.ts:26]: should we have `errorReturnUndefined` true herE?

**Review by aforaleka — COMMENTED** [review]
left a question re errorReturnUndefined in the new function! i think most of these use cases we can probably swallow the error and fail with empty data instead of throwing it

File: src/lib/asyncUtils.ts (hunk 3 above)
Comment (tyleroooo) [reply src/lib/asyncUtils.ts:7]: Yes but in new react-query returning undefined causes a runtime error with no metadata, so we might as well re-throw the error that actually has metadata.

**Comment by tyleroooo** [comment]
I tested a bit just to confirm but it seems like throwing the error has the same behavior now but gives much more descriptive error messages, should make things easier to trace and debug.

**Comment by tyleroooo** [comment]
@aforaleka I made them all return undefined and not re-throw just to be as similar to previous behavior as we can.

**Review by aforaleka — APPROVED** [review]

**Pull request merged**